apps/examples/testcase/le_tc/network/tc_net_igmp.c

```diff
@@ -0,0 +1,363 @@
+#endif
+
+static struct netif test_netif;
+static ip_addr_t test_ipaddr, test_netmask, test_gw;
+struct eth_addr test_ethaddr = {{ 1, 1, 1, 1, 1, 1 }};
+
+static int linkoutput_ctr;
+
+/**
+* @fn					: etharp_remove_all
+* @brief				: Helper function
+* @scenario				:
+* @API's covered		:
+* @Preconditions		:
+* @Postconditions		:
+*/
+static void etharp_remove_all(void)
+{
+	int i;
+	/* call etharp_tmr often enough to have all entries cleaned */
+	for (i = 0; i < 0xff; i++) {
+		etharp_tmr();
+	}
+}
+
+/**
+* @fn					: default_netif_linkoutput
+* @brief				: Helper function
+* @scenario				:
+* @API's covered		:
+* @Preconditions		:
+* @Postconditions		:
+*/
+static err_t default_netif_linkoutput(struct netif *netif, struct pbuf *p)
+{
+	linkoutput_ctr++;
+	return ERR_OK;
+}
+
+/**
+* @fn					: default_netif_init
+* @brief				:
+* @scenario				:
+* @API's covered		:
+* @Preconditions		:
+* @Postconditions		:
+*/
+static err_t default_netif_init(struct netif *netif)
+{
+	netif->linkoutput = default_netif_linkoutput;
+	netif->output = etharp_output;
+	netif->mtu = 1500;
+	netif->flags = NETIF_FLAG_BROADCAST | NETIF_FLAG_ETHARP | NETIF_FLAG_LINK_UP;
+	netif->hwaddr_len = ETHARP_HWADDR_LEN;
+	return ERR_OK;
+}
+
+/**
+* @testcase				: default_netif_add
+* @brief				:
+* @scenario				:
+* @apicovered			: netif_set_default
+* @precondition			:
+* @postcondition		:
+*/
+static void default_netif_add(void)
+{
+	IP4_ADDR(&test_gw, 192, 168, 0, 1);
+	IP4_ADDR(&test_ipaddr, 192, 168, 0, 1);
+	IP4_ADDR(&test_netmask, 255, 255, 0, 0);
+	netif_set_default(netif_add(&test_netif, &test_ipaddr, &test_netmask, &test_gw, NULL, default_netif_init, NULL));
+	netif_set_up(&test_netif);
+}
+
+/**
+* @testcase				:
+* @brief				:
+* @scenario				:
+* @apicovered			: netif_remove
+* @precondition			:
+* @postcondition		:
+*/
+static void default_netif_remove(void)
+{
+	netif_remove(&test_netif);
+}
+
+/**
+* @testcase				: etharp_setup
+* @brief				:
+* @scenario				:
+* @apicovered			: etharp_remove_all
+* @precondition			:
+* @postcondition		:
+*/
+static void etharp_setup(void)
+{
+	etharp_remove_all();
+	default_netif_add();
+}
+
+/**
+* @testcase				: etharp_teardown
+* @brief				:
+* @scenario				:
+* @apicovered			: etharp_remove_all
+* @precondition			:
+* @postcondition		:
+*/
+static void etharp_teardown(void)
+{
+	etharp_remove_all();
+	default_netif_remove();
+}
+
+
+/**
+* @testcase				: tc_etharp_free_entry
+* @brief				:
+* @scenario				:
+* @apicovered			: etharp_free_entry
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_etharp_free_entry(void)
+{
+	int i =1;
+
+	etharp_free_entry(i);
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+* @testcasu				: tc_free_etharp_q
+* @brief				:
+* @scenario				:
+* @apicovered			: free_etharp_q
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_free_etharp_q(void)
+{
+	struct etharp_q_entry *q;
+	q = (struct etharp_q_entry *) malloc (sizeof (struct etharp_q_entry));
+
+	free_etharp_q(q);
```

Comment:
Where is this function? In os/net/lwip/src/netif/etharp.c, there is. But it is a static function. How do you call it in here?